public/JSONtoHTML.coffee

```diff
@@ -0,0 +1,211 @@
+# Variables
+xmlHttp = new XMLHttpRequest
+jsonFile = 'stdLib_base_test.json'
+
+getMainData = (data) ->
+  output = ''
+  modules = data.modules
+
+  i = 0
+  while i < modules.length
+    module = modules[i]
+    output += '<div class="moduleDiv" id="'+ module.name + '">'
+    output += '<div style="width:100%" onMouseOver="document.getElementById(\''+module.name+'_Hover\').style.display = \'inline-block\'" onMouseOut="document.getElementById(\''+module.name+'_Hover\').style.display = \'none\'"><p class="module">' + module.name + '</p>'
+    if module.tag != null && module.tag != undefined 
+      tagType = module.tag.split(":")
+      output += '<div class="tag tag_' + tagType[0].replace(/\s/g,'') + '"><p class="<!--fa fa-tags-->">' + tagType[0].replace(/\s/g,'') + ' in ' + tagType[1].replace(/\s/g,'')+ '</p></div>'
+    output += '<div id="'+module.name+'_Hover" class="hoverableAnchors"><a href="#' + module.name + '" class="fa fa-anchor" title="Get anchor link"></a>&nbsp;&nbsp;&nbsp;&nbsp;<a href="" class="fa fa-file-code" title="Jump to source"></a></div>'
```

Comment:
Can't we use string interpolation here instread of all the "+" operators? Btw @kustosz is it a good way to just glue all the HTML parts, without using some kind of a "template engine" ?

Reply:
but it is faster to connect strings with + than to interpolate them